File: examples/akka01/src/main/java/example/actor/PingActor.java
package example.actor;

import akka.actor.*;
import akka.event.*;

public class PingActor extends UntypedActor{
    private final LoggingAdapter log = Logging.getLogger(getContext().system(), this);
    private ActorRef pongActor;

    public PingActor() {
        this.pongActor = getContext().actorOf(Props.create(PongActor.class), "pongActor");
    }

    public void onReceive(Object message) throws Throwable {
        if (message instanceof String) {
            String msg = (String) message;
            log.info("PingActor received message : {}", msg);
            if("start".equals(msg)) {
                pongActor.tell("ping", getSelf());
            } else if ("pong".equals(msg)) {
                sender().tell("ping", getSelf());
            }
        } else {
            unhandled(message);
        }

    }
}

File: examples/scalaakka01/src/main/scala-2.12/study/p01/actor/PingActor.scala
package study.p01.actor

import akka.actor.{Actor, Props}
import akka.event.Logging

case object PingActor

class PingActor extends Actor {
  val log = Logging(context.system, this)

  val pong = context.actorOf(Props[PongActor], "pongActor")
  def receive = {
    case "start"  => {
      log info "Ping received start"
      pong ! "ping"
    }
    case "pong"   => {
      log info "Ping received pong"
      sender ! "ping"
    }
  }
}
